### PR TITLE
fix: ebs volume restore failure since incorrect unfinished volumes

### LINKED ISSUE #39974: br: ebs restore failure when aws api create volume is not able to return immediately.
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)
5% reproducible
follow https://docs.pingcap.com/zh/tidb-in-kubernetes/v1.4/restore-from-aws-s3-by-snapshot
1. EBS backup 
2. EBS restore

### 2. What did you expect to see? (Required)
restore shall always succeed.
### 3. What did you see instead (Required)
large scale size of tikv node could cause restore failure sometimes.

### 4. What is your TiDB version? (Required)
v6.5.0-pre

### PR DESCRIPTION
**Pull request by fengou1**
### What problem does this PR solve?
Issue Number: close #39974

Problem Summary:

### What is changed and how it works?
when wait to create volume finish, incorrect unfinished volume append, cause 1st restore may failure sometimes.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by YuJuncen — APPROVED** [review]

File: br/pkg/aws/ebs_test.go [hunk 1]
```diff
@@ -14,28 +14,60 @@
...
 import (
+	"context"
 	"testing"
 
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/pingcap/tidb/br/pkg/gluetidb"
 	"github.com/stretchr/testify/require"
 )
 
+func str2Ptr(str string) *string {
```
Comment (3pointer) [inline br/pkg/aws/ebs_test.go:25]: maybe use `aws.String` instead of this function.

File: br/pkg/aws/ebs_test.go [hunk 2]
```diff
@@ -14,28 +14,60 @@
...
+		},
+	}
+
+	mockGlue := &gluetidb.MockGlue{}
+	ctx, _ := context.WithCancel(context.Background())
+	fakeProgress := mockGlue.StartProgress(ctx, "Restore Data", int64(5), false)
+
+	e := &EC2Session{}
+	createdVolumeSize, unfinishedVolumes := e.HandleDescribeVolumesRespose(curentVolumesStates, fakeProgress)
+	require.Equal(t, 4, createdVolumeSize)
```
Comment (3pointer) [inline br/pkg/aws/ebs_test.go:71]: why createdVolumeSize be 4? it seems size is 0

File: br/pkg/aws/ebs_test.go [hunk 3]
```diff
@@ -14,28 +14,60 @@
...
+	require.Equal(t, 1, unfinishedVolumes)
```
Comment (3pointer) [inline br/pkg/aws/ebs_test.go:72]: []*string cannot equal to 1, wrong types

File: br/pkg/aws/ebs_test.go (hunk 1 above)
Comment (fengou1) [reply br/pkg/aws/ebs_test.go]: fix in https://github.com/pingcap/tidb/pull/39975/commits/6017b17dd8e1df611ba031638a6598359d2c8edf

File: br/pkg/aws/ebs_test.go (hunk 2 above)
Comment (fengou1) [reply br/pkg/aws/ebs_test.go:68]: fix in https://github.com/pingcap/tidb/pull/39975/commits/6017b17dd8e1df611ba031638a6598359d2c8edf

File: br/pkg/aws/ebs_test.go (hunk 3 above)
Comment (fengou1) [reply br/pkg/aws/ebs_test.go]: fix in https://github.com/pingcap/tidb/pull/39975/commits/6017b17dd8e1df611ba031638a6598359d2c8edf

**Review by 3pointer — APPROVED** [review]

**Pull request merged**